x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_logic.ts

```diff
@@ -13,7 +13,7 @@ import { EngineDetails } from './types';
 
 interface EngineValues {
   dataLoading: boolean;
-  engine: EngineDetails | {};
+  engine: Partial<EngineDetails>;
```

Comment:
Using `EngineDetails | {}` can cause some tough issues to solve when used. 

Specifically, if you try to access properties of engine you receive an error message.

```
engine.apiKey
```

```
Property 'apiKey' does not exist on type '{} | EngineDetails'.
  Property 'apiKey' does not exist on type '{}'
```

I'm actually not sure how to work around this.

Reply:
For the sake of this PR, `Partial<SomeType>` is what we use elsewhere, so this at least moves us through to consistent and passes the type check. However, we might want to think about our approach more holistically in a separate PR.

Reply:
I've personally been running into the exact same type issues with ` | {}` myself and `Partial<>` just seems so much more friendly 😬 That being said I'm not a Typescript expert/purist of course, definitely happy to revisit